apps/backoffice-v2/src/pages/Entity/hooks/useTasks/hooks/useDirectorsBlocks/useDirectorsBlocks.tsx

```diff
@@ -86,6 +78,97 @@ export const useDirectorsBlocks = (
+                      props: {
+                        ...motionProps,
+                        variant: 'warning',
+                        className: 'min-h-8 text-sm font-bold',
+                      },
+                    },
+                  ]
+                : [
+                    {
+                      type: 'badge',
+                      value: (
+                        <React.Fragment>
+                          Re-upload needed
+                          <X
+                            className="h-4 w-4 cursor-pointer"
+                            onClick={() =>
+                              removeDecisionById({
+                                documentId: document.id,
+                                contextUpdateMethod: 'director',
+                              })
+                            }
+                          />
+                        </React.Fragment>
+                      ),
+                      props: {
+                        ...motionProps,
+                        variant: 'warning',
+                        className: `gap-x-1 min-h-8 text-white bg-warning text-sm font-bold`,
+                      },
+                    },
+                  ]
+              : []),
+            ...(!isApproved && !isRevision
+              ? [
+                  {
+                    type: 'callToAction',
+                    value: {
+                      text: 'Re-upload needed',
+                      props: {
+                        revisionReasons: getRevisionReasonsForDocument(document, workflow),
+                        disabled:
+                          (!isDoneWithRevision && Boolean(document.decision?.status)) || noAction,
+                        decision: 'reject',
+                        id: document.id,
+                        contextUpdateMethod: 'director',
+                      },
+                    },
+                  },
+                ]
+              : []),
+            ...(isApproved
+              ? [
+                  {
+                    type: 'badge',
+                    value: 'Approved',
+                    props: {
+                      ...motionProps,
+                      variant: 'success',
+                      className: `text-sm min-h-8 font-bold bg-success/20`,
+                    },
+                  },
+                ]
+              : !isRevision
+              ? [
+                  {
+                    type: 'callToAction',
+                    value: {
+                      text: 'Approve',
+                      props: {
+                        decision: 'approve',
+                        id: document.id,
+                        contextUpdateMethod: 'director',
+                        disabled:
+                          (!isDoneWithRevision && Boolean(document?.decision?.status)) || noAction,
+                      },
+                    },
+                  },
+                ]
+              : []),
+          ];
```

Comment:
Can we refactor these spreads + ternaries to `if` statements instead? Seems like we have flows based on the state and that may also change over time.